### PR TITLE
Hotswap: Add option to sponsor transaction

### PR DESCRIPTION
**Pull request by cuongph87**

### CODE REVIEW

File: lib/src/sep/0099/hotswap.dart [hunk 1]
```diff
@@ -106,12 +118,21 @@ class Hotswap {
       '0',
     ).build();
 
+    EndSponsoringFutureReservesOperation endSponsorshipOperation =
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply lib/src/sep/0099/hotswap.dart:121]: Done.

File: lib/src/sep/0099/hotswap.dart [hunk 2]
```diff
@@ -69,11 +69,23 @@ class Hotswap {
...
+    sponsored ??= true;
+    if (sponsored && sponsoringAccountId == null) {
+      throw Exception(
+          'Sponsoring account ID is required for sponsored transaction.');
+    }
     var account = await _sdk.accounts.account(accountId);
 
     final txBuilder = TransactionBuilder(account);
 
+    BeginSponsoringFutureReservesOperation beginSponsoringOperation =
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply lib/src/sep/0099/hotswap.dart:84]: Done.

File: lib/src/sep/0099/hotswap.dart [hunk 3]
```diff
@@ -69,11 +69,23 @@ class Hotswap {
...
+        BeginSponsoringFutureReservesOperationBuilder(accountId)
+            .setSourceAccount(sponsoringAccountId!)
+            .build();
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply lib/src/sep/0099/hotswap.dart:87]: Done.

File: lib/src/sep/0099/hotswap.dart [hunk 4]
```diff
@@ -69,11 +69,23 @@ class Hotswap {
     required String accountId,
     required HotswapRoute hotswapRoute,
     String toAssetTrustLineLimit = ChangeTrustOperationBuilder.MAX_LIMIT,
+    bool? sponsored,
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply lib/src/sep/0099/hotswap.dart:72]: Done.

**Review by jopmiddelkamp — APPROVED** [review]

**Pull request merged**